backtesting/_plotting.py

```diff
@@ -537,10 +537,22 @@ def __eq__(self, other):
             colors = value._opts['color']
             colors = colors and cycle(_as_list(colors)) or (
                 cycle([next(ohlc_colors)]) if is_overlay else colorgen())
-            legend_label = LegendStr(value.name)
-            for j, arr in enumerate(value, 1):
+
+            tooltip_label = value.name if isinstance(value.name, str) else ", ".join(value.name)
+
+            if len(value) == 1:
+                legend_labels = [LegendStr(item) for item in _as_list(value.name)]
+            elif isinstance(value.name, str):
+                legend_labels = [
+                    LegendStr(f"{name}[{index}]")
+                    for index, name in enumerate(repeat(value.name, len(value)))
+                ]
+            else:
+                legend_labels = [LegendStr(item) for item in value.name]
```

Comment:
```suggestion
            if isinstance(value.name, str):
                legend_labels = [
                    LegendStr(f"{name}[{index}]")
                    for index, name in enumerate(repeat(value.name, len(value)))
                ]
            else:
                legend_labels = [LegendStr(item) for item in _as_list(value.name)]
```

Reply:
Hmm, this does not seem to be correct :thinking: 

There are three use cases:
 * The indicator returns one value, one name `name` is passed. The label will be `name`.
 * The indicator returns multiple values, one name `name` is passed. The labels will be `name[0]`, `name[1]`, ...
 * The indicator returns multiple values, multiple names `name_one`, `name_two`, ... are passed. The labels will be the same as names.

There are three branches exactly to account for each use case. Now, if we merge this suggestion, the first use case will be broken, as the label will be `name[0]`. Am I missing something?

Reply:
I think I understand your point: there are two very similar branches. I got rid of one of them, and my scenarios above still work. Please, have a look.